gpflow/conditionals.py

```diff
@@ -140,30 +140,25 @@ def _sample_conditional(Xnew, feat, kern, f, *, full_cov=False, full_output_cov=
     However, for some combinations of Mok and Mof more efficient sampling routines exists.
     The dispatcher will make sure that we use the most efficient one.
 
-    :return: N x P (full_output_cov = False) or N x P x P (full_output_cov = True)
+    :return: (S x) N x (P/D)
     """
+    if full_cov and full_output_cov:
+        raise NotImplementedError("The combination of both full_cov and full_output_cov is not "
+                                  "implemented for sample_conditional.")
+
     logger.debug("sample conditional: InducingFeature Kernel")
-    mean, var = conditional(Xnew, feat, kern, f, full_cov=full_cov, full_output_cov=full_output_cov,
-                            q_sqrt=q_sqrt, white=white)  # N x P, N x P (x P)
+    mean, var = conditional(Xnew, feat, kern, f, q_sqrt=q_sqrt, white=white,
+                            full_cov=full_cov, full_output_cov=full_output_cov)
     if full_cov:
-        if full_output_cov:
-            raise NotImplementedError("The combination of both full_cov and full_output_cov is not "
-                                      "implemented for sample_conditional.")
-        N, D = tf.shape(mean)[0], tf.shape(mean)[1]
-        S = num_samples
-        jittermat = settings.numerics.jitter_level * tf.eye(N, batch_shape=[D], dtype=settings.float_type)  # [D, N, N]
-        L = tf.cholesky(var + jittermat)  # [D, N, N]
-        nu = tf.random_normal([D, N, S], dtype=float_type)  # [D, N, S]
-        samples = mean[..., None] + tf.transpose(tf.matmul(L, nu), [1, 0, 2])  # [N, D, S]
+        # mean: N x P
+        # var: P x N x N
+        mean = tf.transpose(mean, [0, 1])  # now P x N
+        samples = _sample_mvn(mean, var, 'full', num_samples=num_samples)  # (S x) P x N
+        samples = tf.transpose(samples, [-1, -2])  # now (S x) N x P
```

Comment:
`matrix_transpose` here will do

Reply:
thanks... the first one (two lines above) was also wrong :/ fixed now.